x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/ListEnginesAction.java

```diff
@@ -0,0 +1,175 @@
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.action.util.PageParams;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+import org.elasticsearch.xpack.entsearch.engine.EngineIndexService;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class ListEnginesAction extends ActionType<ListEnginesAction.Response> {
+
+    public static final ListEnginesAction INSTANCE = new ListEnginesAction();
+    public static final String NAME = "indices:admin/engine/list";
+
+    public ListEnginesAction() {
+        super(NAME, ListEnginesAction.Response::new);
+    }
+
+    // TODO Check if this should implement IndicesRequest.Replaceable
+    public static class Request extends ActionRequest implements IndicesRequest {
+
+        private static final String DEFAULT_QUERY = "*";
+        private final String query;
+        private final PageParams pageParams;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.query = Objects.requireNonNullElse(in.readOptionalString(), DEFAULT_QUERY);
+            this.pageParams = new PageParams(in);
+        }
+
+        public Request(@Nullable String query, PageParams pageParams) {
+            this.query = Objects.requireNonNullElse(query, DEFAULT_QUERY);
+            this.pageParams = pageParams;
+        }
+
+        public String query() {
+            return query;
+        }
+
+        public PageParams pageParams() {
+            return pageParams;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            // Pagination validation is done as part of PageParams constructor
+            ActionRequestValidationException validationException = null;
+            if (Strings.isEmpty(query())) {
+                validationException = ValidateActions.addValidationError("Engine query is missing", validationException);
+            }
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeOptionalString(query);
+            pageParams.writeTo(out);
+        }
+
+        @Override
+        public String[] indices() {
+            return new String[] { EngineIndexService.ENGINE_ALIAS_NAME };
```

Comment:
@carlosdelest I think this needs to be the engine alias and that's why the yaml tests are failing.

Reply:
This is pointing already to the engine index alias name. Do you mean to the individual engines indices?

Reply:
This won't work since we don't have any index name to pass here (it's a list API). I guess that means that we need to add a new cluster privilege to manage Engine? @ioanatia what's your take on this? We assumed that using the index privilege was enough for now but that won't work for this API. Maybe we can start with reusing the cluster monitor privilege for this API and continue the discussion async in order to not block this PR?